src/components/para_trials/taskCard.tsx

```diff
@@ -1,48 +1,55 @@
 // import Image from "next/image";
-import React, { useState } from "react";
-import styles from "./styles/TaskCard.module.css";
+import React, { useEffect, useState } from "react";
+import $taskCard from "./styles/TaskCard.module.css";
+import $button from "@/styles/Button.module.css";
+import $box from "@/styles/Box.module.css";
 import Link from "next/link";
 import ProgressBar from "./progressBar";
 import { ParaTaskCard } from "@/types/global";
+import { differenceInWeeks, format } from "date-fns";
 
 interface TaskCardProps {
   task: ParaTaskCard;
 }
 
 const TaskCard = ({ task }: TaskCardProps) => {
-  // TODO: calculate completion rate depending on trials
-  const [completionRate] = useState(0);
+  const [completionRate, setCompletionRate] = useState(0);
+  useEffect(() => {
+    const calculatedRate = Math.floor(
+      ((task.completed_trials || 0) / task.trial_count) * 100
+    );
+    setCompletionRate(calculatedRate);
+  }, [task]);
```

Comment:
`useEffect()` should be avoided when possible as it can result in unnecessary renders, I think `useMemo()` would work here instead?